client/starwhale/core/dataset/type.py

```diff
@@ -659,14 +659,20 @@ def __init__(
         )
 
     def to_bytes(self, encoding: str = "") -> bytes:
+        self.link_to_content(encoding)
         return self.content.encode(encoding or self.encoding)
 
     def to_numpy(self) -> numpy.ndarray:
         return numpy.array(self.to_str(), dtype=self.dtype)
 
-    def to_str(self) -> str:
+    def to_str(self, encoding: str = "") -> str:
+        self.link_to_content(encoding)
         return self.content
 
+    def link_to_content(self, encoding: str = "") -> None:
```

Comment:
move these logical into line649 in the __init__ function? so we can get a unique self.content in the whole Text class.

Reply:
links are used for lazy initialing. If content in link is fetched in init function, the content will be stored into datastore directlly.

Reply:
ok.
for the lazy loading, we may use the following schema:

```python
from functools import cached_property

def __init__(self, content):
   self._content = content

@cached_property
def content(self):
   if not self._content:
       self._content = self.link_to_content()

   return self._content

...

call `self.content` property 
````

Reply:
> got.
> 
> for the lazy loading, we may use the following schema:
> 
> ```python
> from functools import cached_property
> 
> def __init__(self, content):
>    self._content = content
> 
> @cached_property
> def content(self):
>    if not self._content:
>        self._content = self.link_to_content()
> 
>    return self._content
> 
> ...
> 
> call `self.content` property in `to_bytes` or `to_str` function.
> ```

OK

Reply:
done